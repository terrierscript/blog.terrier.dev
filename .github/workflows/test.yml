name: Test

on: [push]

jobs:
  build-test:
    runs-on: ubuntu-latest
    steps:
    - uses: actions/checkout@v1
    - run: yarn install
    - run: yarn test
  tsc:
    runs-on: ubuntu-latest
    steps:
    - uses: actions/checkout@v1
    - run: yarn install
    - run: yarn tsc
  jest:
    runs-on: ubuntu-latest
    steps:
    - uses: actions/checkout@v1
    - run: yarn install
    - run: yarn jest
  lhci:
    runs-on: ubuntu-latest
    steps:
    - uses: actions/checkout@v1
    - run: yarn install
    - run: yarn build
    - name: Audit URLs using Lighthouse
      uses: treosh/lighthouse-ci-action@v2
      with:
        staticDistDir: "./dist"
    - name: Save results
      uses: actions/upload-artifact@v1
      with:
        name: lighthouse-results
        path: '.lighthouseci'
