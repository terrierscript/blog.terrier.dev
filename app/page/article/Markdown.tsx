import React, { FC } from "react"
import ReactMarkdown from "react-markdown"
import { Prism as SyntaxHighlighter } from "react-syntax-highlighter"
import style from "react-syntax-highlighter/dist/cjs/styles/prism/tomorrow"
import { nl2brRemark } from "./unified/nl2br"

export const CodeBlock = ({ value, language }) => {
  return (
    <SyntaxHighlighter language={language} style={style}>
      {value}
    </SyntaxHighlighter>
  )
}

export const RawMarkdown: FC<{ markdown: string }> = ({ markdown }) => {
  return (
    <ReactMarkdown
      source={markdown}
      escapeHtml={false} // for web components
      renderers={{ code: CodeBlock }}
      plugins={[nl2brRemark]}
    />
  )
}
