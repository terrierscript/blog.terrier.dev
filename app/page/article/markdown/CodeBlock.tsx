import React from "react"
import { Prism as SyntaxHighlighter } from "react-syntax-highlighter"
import style from "react-syntax-highlighter/dist/cjs/styles/prism/tomorrow"

export const PrismCodeBlock = ({ value, language }) => {
  return (
    <SyntaxHighlighter language={language} style={style}>
      {value}
    </SyntaxHighlighter>
  )
}
