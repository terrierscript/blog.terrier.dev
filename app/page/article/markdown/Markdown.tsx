import React, { FC } from "react"
import ReactMarkdown from "react-markdown/with-html" // for web components
import { nl2brRemark } from "./unified/nl2br"
import styled from "@emotion/styled"
import { PrismCodeBlock } from "./CodeBlock"
import {
  ARTICLE_COLOR,
  // TITLE_COLOR,
  ARTICLE_TITLE_COLOR
  // FEED_LINK_COLOR
} from "../../../layout/global/colors"

export const CodeWrapper = styled.div`
  margin-bottom: 1.8em;
`

const Link = styled.a`
  text-decoration: underline;
`
const ArticleWrapper = styled.div`
  font-size: 0.9em;
  color: ${ARTICLE_COLOR};
  line-height: 1.3em;
  h1,
  h2,
  h3,
  h4,
  h5,
  h6 {
    margin: 1em 0;
    color: ${ARTICLE_TITLE_COLOR};
  }
`

const CodeBlock = props => (
  <CodeWrapper>
    <PrismCodeBlock {...props} />
  </CodeWrapper>
)

const List = styled.ul`
  margin-bottom: 0.25em;
`
const ListItem = styled.li`
  margin-top: 0.25em;
`
const Paragraph = styled.p`
  margin-bottom: 0.8em;
`

const InlineCode = styled.span`
  border: 1px solid #ccc;
  background: #eee;
  border-radius: 4px;
  padding: 0.1em 0.4em;
  margin: 0.1em;
`

export const RawMarkdown: FC<{ markdown: string }> = ({ markdown }) => {
  return (
    <ArticleWrapper>
      <ReactMarkdown
        source={markdown}
        escapeHtml={false} // for web components
        renderers={{
          paragraph: Paragraph,
          code: CodeBlock,
          link: Link,
          list: List,
          listItem: ListItem,
          inlineCode: InlineCode
        }}
        plugins={[nl2brRemark]}
      />
    </ArticleWrapper>
  )
}
