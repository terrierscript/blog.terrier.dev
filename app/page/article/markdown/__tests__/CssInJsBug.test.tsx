import React from "react"
import renderer from "react-test-renderer"
import { RawMarkdown } from "../Markdown"
import { PrismCodeBlock } from "../CodeBlock"
import { removeIndent, wrapCode } from "./Markdown.test"

describe("Syntax Highlight", () => {
  it("css-in-js bug", () => {
    const code = `
    const Item = styled.div\`
      grid-row: ${({ row }) => row};
      grid-columns: ${({ col }) => col};
    \``
    const tree = renderer
      .create(<PrismCodeBlock value={code} language={"tsx"} />)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})
