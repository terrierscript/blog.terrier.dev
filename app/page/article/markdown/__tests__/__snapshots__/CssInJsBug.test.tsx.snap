// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Syntax Highlight css-in-js bug 1`] = `
<pre
  style={
    Object {
      "MozHyphens": "none",
      "MozTabSize": "4",
      "OTabSize": "4",
      "WebkitHyphens": "none",
      "background": "#2d2d2d",
      "color": "#ccc",
      "fontFamily": "Consolas, Monaco, 'Andale Mono', 'Ubuntu Mono', monospace",
      "hyphens": "none",
      "lineHeight": "1.5",
      "margin": ".5em 0",
      "msHyphens": "none",
      "overflow": "auto",
      "padding": "1em",
      "tabSize": "4",
      "textAlign": "left",
      "whiteSpace": "pre",
      "wordBreak": "normal",
      "wordSpacing": "normal",
      "wordWrap": "normal",
    }
  }
>
  <code
    style={
      Object {
        "MozHyphens": "none",
        "MozTabSize": "4",
        "OTabSize": "4",
        "WebkitHyphens": "none",
        "background": "none",
        "color": "#ccc",
        "fontFamily": "Consolas, Monaco, 'Andale Mono', 'Ubuntu Mono', monospace",
        "hyphens": "none",
        "lineHeight": "1.5",
        "msHyphens": "none",
        "tabSize": "4",
        "textAlign": "left",
        "whiteSpace": "pre",
        "wordBreak": "normal",
        "wordSpacing": "normal",
        "wordWrap": "normal",
      }
    }
  >
    
    
    <span
      className="token"
      style={
        Object {
          "color": "#cc99cd",
        }
      }
    >
      const
    </span>
     
    <span
      className="token maybe-class-name"
      style={Object {}}
    >
      Item
    </span>
     
    <span
      className="token"
      style={
        Object {
          "color": "#67cdcc",
        }
      }
    >
      =
    </span>
     styled
    <span
      className="token"
      style={
        Object {
          "color": "#ccc",
        }
      }
    >
      .
    </span>
    <span
      className="token property-access"
      style={Object {}}
    >
      div
    </span>
    <span
      className="token template-string"
      style={Object {}}
    >
      <span
        className="token template-punctuation"
        style={
          Object {
            "color": "#7ec699",
          }
        }
      >
        \`
      </span>
      <span
        className="token css language-css"
        style={Object {}}
      >
        
      
        <span
          className="token"
          style={
            Object {
              "color": "#cc99cd",
            }
          }
        >
          grid-row: function 
          <span
            className="token"
            style={
              Object {
                "color": "#ccc",
              }
            }
          >
            (
          </span>
          _ref
          <span
            className="token"
            style={
              Object {
                "color": "#ccc",
              }
            }
          >
            )
          </span>
        </span>
         
        <span
          className="token"
          style={
            Object {
              "color": "#ccc",
            }
          }
        >
          {
        </span>
        
      var row = _ref
        <span
          className="token"
          style={
            Object {
              "color": "#f08d49",
            }
          }
        >
          .
        </span>
        row
        <span
          className="token"
          style={
            Object {
              "color": "#ccc",
            }
          }
        >
          ;
        </span>
        
      return row
        <span
          className="token"
          style={
            Object {
              "color": "#ccc",
            }
          }
        >
          ;
        </span>
        
    
        <span
          className="token"
          style={
            Object {
              "color": "#ccc",
            }
          }
        >
          }
        </span>
        <span
          className="token"
          style={
            Object {
              "color": "#cc99cd",
            }
          }
        >
          ;
      grid-columns: function 
          <span
            className="token"
            style={
              Object {
                "color": "#ccc",
              }
            }
          >
            (
          </span>
          _ref2
          <span
            className="token"
            style={
              Object {
                "color": "#ccc",
              }
            }
          >
            )
          </span>
        </span>
         
        <span
          className="token"
          style={
            Object {
              "color": "#ccc",
            }
          }
        >
          {
        </span>
        
      var col = _ref
        <span
          className="token"
          style={
            Object {
              "color": "#f08d49",
            }
          }
        >
          2.
        </span>
        col
        <span
          className="token"
          style={
            Object {
              "color": "#ccc",
            }
          }
        >
          ;
        </span>
        
      return col
        <span
          className="token"
          style={
            Object {
              "color": "#ccc",
            }
          }
        >
          ;
        </span>
        
    
        <span
          className="token"
          style={
            Object {
              "color": "#ccc",
            }
          }
        >
          }
        </span>
        <span
          className="token"
          style={
            Object {
              "color": "#ccc",
            }
          }
        >
          ;
        </span>
        
    
      </span>
      <span
        className="token template-punctuation"
        style={
          Object {
            "color": "#7ec699",
          }
        }
      >
        \`
      </span>
    </span>
  </code>
</pre>
`;
